sdk/file_sdk/XMPFileFormat/xmp_file_format.cpp

```diff
@@ -1,15 +1,169 @@
+    auto siteId = GetStringByKey(metadata, labelsItemPath, "SiteId", "Ref");
+    bool enabled;
+
+    TryParseBool(GetStringByKey(metadata, labelsItemPath, "Enabled", nullptr), enabled);
+
+    auto parentId = GetStringByKey(metadata, labelsItemPath, "ParentId", nullptr);
+
+    auto owner = GetStringByKey(metadata, labelsItemPath, "Owner", "AssignedBy");
+    if (owner == "")
+      owner = GetStringByKey(metadata, labelsItemPath, "Owner", "SetBy");
+
+    auto setDate = GetStringByKey(metadata, labelsItemPath, "SetDate", "DateCreated");
+    auto application = GetStringByKey(metadata, labelsItemPath, "Application", nullptr);
+
+    std::string path, method;
+    SXMPUtils::ComposeStructFieldPath(kMsipNamespace, labelsItemPath.c_str(), kMsipLabelExtendedNamespace, "MSFT_Method", &path);
+    metadata->GetProperty(kMsipNamespace, path.c_str(), &method, 0);
+
+    Method methodEnum = Method::NONE;
+    if(EqualsIgnoreCase(method, "automatic"))
+      methodEnum = Method::AUTOMATIC;
+    else if(EqualsIgnoreCase(method, "manual"))
+      methodEnum = Method::MANUAL;
+
+    std::string propVendor, propKey, propValue, extendedStructPath;
+    bool exists = true;
+
+    vector<ExtendedProperty> properties;
+    exists = metadata->DoesStructFieldExist(kMsipNamespace, labelsItemPath.c_str(), kMsipLabelNamespace, kExtendedStructName);
+    if (exists)
+    {
+      SXMPUtils::ComposeStructFieldPath(kMsipLabelNamespace, labelsItemPath.c_str(), kMsipLabelNamespace, kExtendedStructName, &extendedStructPath);
+      XMP_Index propCount = metadata->CountArrayItems(kMsipNamespace, extendedStructPath.c_str());
+      for (int j = 1; j <= propCount; j++)
+      {
+        SXMPUtils::ComposeArrayItemPath(kMsipNamespace, extendedStructPath.c_str(), j, &labelsItemPath);
+
+        SXMPUtils::ComposeStructFieldPath(kMsipNamespace, labelsItemPath.c_str(), kMsipLabelExtendedNamespace, "Vendor", &path);
+        metadata->GetProperty(kMsipNamespace, path.c_str(), &propVendor, 0);
+
+        SXMPUtils::ComposeStructFieldPath(kMsipNamespace, labelsItemPath.c_str(), kMsipLabelExtendedNamespace, "Key", &path);
+        metadata->GetProperty(kMsipNamespace, path.c_str(), &propKey, 0);
+
+        SXMPUtils::ComposeStructFieldPath(kMsipNamespace, labelsItemPath.c_str(), kMsipLabelExtendedNamespace, "Value", &path);
+        metadata->GetProperty(kMsipNamespace, path.c_str(), &propValue, 0);
+
+        ExtendedProperty property;
+        property.vendor = propVendor;
+        property.key = propKey;
+        property.value = propValue;
+
+        properties.push_back(property);
+      }
+    }
+
+    tags.push_back(Tag(labelId, labelName, parentId, owner, enabled, setDate, application, methodEnum, siteId, properties));
+  }
+
+  return tags;
+}
+
+static void Initialize()
+{
+  if (Initialized)
+    return;
+
+  InitMutex.lock();
+  if (Initialized)
+    return;
+
+  if (!SXMPMeta::Initialize())
+    new std::runtime_error("SXMPMeta Error");
```

Comment:
don't you need to throw it?

Reply:
Sure, i didn't notice i forgot to add throw